aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -709,21 +713,56 @@ private void launchSignOut(final String redirectUri, final String browserPackage
         launchCustomTabs(fqdn, null, browserPackage);
     }
 
+    /**
+     * Returns a package that support Custom Tabs.
+     * Default Chrome if available, else return first supported package in the list.
+     */
+    private String getSupportedBrowserPackage(){
+        PackageManager packageManager = context.getPackageManager();
+        // Get default VIEW intent handler.
+        Intent activityIntent = new Intent()
+                .setAction(Intent.ACTION_VIEW)
+                .addCategory(Intent.CATEGORY_BROWSABLE)
+                .setData(Uri.fromParts("http", "", null));
+
+        // Get all apps that can handle VIEW intents.
+        List<ResolveInfo> resolvedActivityList = packageManager.queryIntentActivities(activityIntent, 0);
+        ArrayList<String> packageNamesSupportingCustomTabs = new ArrayList<>();
+        for (ResolveInfo info : resolvedActivityList) {
+            Intent serviceIntent = new Intent()
+                .setAction(ACTION_CUSTOM_TABS_CONNECTION)
+                .setPackage(info.activityInfo.packageName);
+            // Check if this package also resolves the Custom Tabs service.
+            if(packageManager.resolveService(serviceIntent, 0) != null) {
+                packageNamesSupportingCustomTabs.add(info.activityInfo.packageName);
+            }
+        }
+
+        if (!packageNamesSupportingCustomTabs.contains(DEFAULT_BROWSER_PACKAGE)) {
+            return packageNamesSupportingCustomTabs.get(0);
+        }
+
+        return DEFAULT_BROWSER_PACKAGE;
+    }
+
     /**
      * Launches the HostedUI webpage on a Custom Tab.
      * @param uri Required: {@link Uri}.
      * @param activity Activity to launch custom tabs from and which will listen for the intent completion.
      * @param browserPackage Optional string specifying the browser package to launch the specified url.
-     *                       Defaults to Chrome if null.
+     *                       Launches intent chooser if set to "customtabs.intent.chooser".
+     *                       Defaults to Chrome(if available, else use the first supported browser in the list) if null.
      */
     private void launchCustomTabs(final Uri uri, final Activity activity, final String browserPackage) {
     	try {
 	        CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder(mCustomTabsSession);
 	        mCustomTabsIntent = builder.build();
 	        if(pool.getCustomTabExtras() != null)
 	            mCustomTabsIntent.intent.putExtras(pool.getCustomTabExtras());
-	        mCustomTabsIntent.intent.setPackage(
-	                browserPackage != null ? browserPackage : DEFAULT_BROWSER_PACKAGE);
+	        if(browserPackage != "customtabs.intent.chooser") {
+	            mCustomTabsIntent.intent.setPackage(
```

Comment:
Do we even need to call `setPackage` at all?  What happens if we don't set it?  Per this custom tabs integration guide, it doesn't actually say you need to set it.   Perhaps if we don't set it, then it will just proceed if one is present, or prompt the user if there are multiple?    Prompting the user if multiple are installed is really the ideal behavior anyway.

Reply:
Yes that's correct. If we don't set the package, it will just use the one set as default. If default is not set then use the browser that is available or prompt the user in case of multiple.